src/ansys/fluent/core/quantity.py

```diff
@@ -0,0 +1,197 @@
+from itertools import permutations
+
+import pint
+from pint import Unit
+
+ureg = pint.UnitRegistry(autoconvert_offset_to_baseunit=True)
+ureg.default_system = "SI"
+
+
+quantity = ureg.Quantity
+
+
+restricted_units = ["Hz", "hertz", "rad/s", "radian/s", "rpm", "rps", "cps"]
```

Comment:
This as a flat list does not cover what we need since it prevents legal conversions also. We would need to structure this into list of pairs of lists.

However, we need to have a solution that scales properly (including as yet unimagined unit systems). The required approach is to compute the dimensions from the units and compare those. However @hpohekar informs me that pint does not include angle in its dimensions definitions. One option is to drop the requirement for this restriction altogether and just live with the pint behaviour.

Reply:
Yes, I agree with this, because ```rad/s, radian/s``` can also be given as ```rad s^-1, radian s^-1```. We can not cover every possible variation of same unit by going this approach.

Reply:
I have also gone through the conversion table from ```units.scm``` and there is no such conversion made between Hz, rad/s.  

![image](https://user-images.githubusercontent.com/106588300/186412966-6ef89187-2546-4f70-9c37-8fa0ff77c8bc.png)

Also by default pint is correctly providing conversion between ```rpm``` and ```rad/s``` which we have restricted above.

![image](https://user-images.githubusercontent.com/106588300/186413912-1d517749-3cc6-4edf-9041-e98b4945d171.png)

![image](https://user-images.githubusercontent.com/106588300/186414086-10c3ac18-e81b-493d-b740-4e06c12cd05e.png)

Hence I think instead of restricting conversions manually it's better to stick with default pint behaviour.

Reply:
Yes, that's fine by me.  @seanpearsonuk maybe submit an issue to the pint repo for discussion?  Interesting how even the internet does not get these details right.  :-).

Reply:
@hpohekar Curious... What was the rationale to restrict converting rpm and radians/s?   `1 rev = 2*pi` radians right, they are dimensionally consistent.

Reply:
Yes, that's right. They are dimensionally consistent. Actually we wanted to restrict conversion between Hz, hertz, rpm but by taking all permutations together it formed this pair. 

![image](https://user-images.githubusercontent.com/106588300/186436192-ec2f9e45-da2d-49b6-a81d-05b60fa9bf4a.png)


![image](https://user-images.githubusercontent.com/106588300/186436044-1d1805a3-ac1f-4664-8435-94f5d3877492.png)

Reply:
Removed restricted code and related test cases and updated the documentation accordingly.